src/app/assumptions-and-limitation/assumptions-and-limitation.component.spec.ts

```diff
@@ -41,26 +40,14 @@ describe('AssumptionsAndLimitationComponent', () => {
     expect(hasFocus).toBeTrue();
   });
 
-  it('should emit close event with true value when press esc key with focus within component', () => {
+  it('should emit close event with { true, true } value when press esc key with focus within component', () => {
     spyOn(component.closeEvent, 'emit');
 
     component.ngAfterContentInit();
     fixture.detectChanges();
 
     document.dispatchEvent(new KeyboardEvent('keydown', { key: 'Escape' }));
 
-    expect(component.closeEvent.emit).toHaveBeenCalledWith(true);
-  });
-
-  it('should emit close event with false value when press esc key with focus not within component', () => {
```

Comment:
This test was testing the opposite scenario to the one in the description. I'm not sure how to drop focus on the component without testing from a parent component.

Reply:
Yeah, I think you'd have to set up a test component to contain it and go through it. You'd also be testing that nothing happens now essentially - if you hit escape with diagram focussed, diagram remains focussed. Could be that's easier to do in Playwright?

Reply:
Yes this one might be best left out of the unit tests. We could add a Playwright test if we would like to test this scenario.